pkg/alertmanager/operator.go

```diff
@@ -1361,18 +1362,32 @@ func checkAlertmanagerRoutes(r *monitoringv1alpha1.Route, receivers map[string]s
 }
 
 // configureHTTPConfigInStore configure the asset store for HTTPConfigs.
-func configureHTTPConfigInStore(ctx context.Context, httpConfig *monitoringv1alpha1.HTTPConfig, namespace string, key string, store *assets.Store) error {
+func configureHTTPConfigInStore(ctx context.Context, httpConfig *monitoringv1alpha1.HTTPConfig, amVersion string, namespace string, key string, store *assets.Store) error {
 	if httpConfig == nil {
 		return nil
 	}
 
+	// Version checks before we configure the http config
 	var err error
+	version, err := semver.ParseTolerant(amVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse alertmanager version")
+	}
+
+	if version.GTE(semver.MustParse("0.22.0")) && httpConfig.Authorization != nil {
+		return errors.Wrap(err, "Authorization is only supported for alertmanager versions greater than v0.22+")
```

Comment:
I would rather see that the operator is liberal here (e.g. doesn't fail Alertmanager resources that define the `Authorization` field while version < v0.22) and strict when it comes to generate the configuration (e.g. only use the `Authorization` when version >= v0.22). This would match what the existing behavior for the Prometheus CRD.

Reply:
I can do the change but there is no logger down the stack of check methods. Should I just return nil here or put a logger in all checks ? This would just reject the alertmanagerconfig as the code is currently doing for an empty username in basic auth

Reply:
@simonpasquier what do you think I should do regarding this ?